src/charm.py

```diff
@@ -27,34 +29,65 @@ def __init__(self, *args: typing.Any):
         super().__init__(*args)
 
         self.falco_layout = FalcoLayout(base_dir=self.charm_dir / "falco")
-        self.falco_service_file = FalcoServiceFile(self.falco_layout)
-        self.managed_falco_config = FalcoConfig(self.falco_layout, self)
+        self.falco_service_file = FalcoServiceFile(self.falco_layout, self)
+        self.managed_falco_config = FalcoConfig(self.falco_layout)
         self.falco_service = FalcoService(self.managed_falco_config, self.falco_service_file)
 
         self.framework.observe(self.on.remove, self._on_remove)
         self.framework.observe(self.on.install, self._on_install_or_upgrade)
         self.framework.observe(self.on.upgrade_charm, self._on_install_or_upgrade)
         self.framework.observe(self.on.update_status, self._update_status)
```

Comment:
update status will be executed periodically. I'm not sure you want to actaully listen to this event